var/spack/repos/builtin/packages/mapl/package.py

```diff
@@ -159,6 +170,10 @@ class Mapl(CMakePackage):
     # Patch to add missing MPI Fortran target to top-level CMakeLists.txt
     patch("mapl-2.12.3-mpi-fortran.patch", when="@:2.12.3")
 
+    # MAPL only compiles with version 2.42.0 and higher so we conflict
+    # with older versions
+    conflicts("mpich@4", when="@:2.41")
```

Comment:
This could cause the concretizer go looking for earlier versions of `mpich` to concretize with. Is it true that earlier versions will work with `mpich@3`?

Reply:
@alecbcs Oooh. Good eye. Okay, how can I say:

1. We only support MPICH 4 (because I've never tested anything else)
2. And then only MPICH 4 after that. Something like:

```python
conflicts("mpich@:3")
conflicts("mpich@4", when="@:2.41")
```

cc: @alalazo (who knows this stuff better than me...or anyone?)

Reply:
@climbfuji Does this seem a bit better to you?

Reply:
> ```python
> conflicts("mpich@:3")
> conflicts("mpich@4", when="@:2.41")
> ```

This looks good to me. @mathomp4 if you add this to the PR I think we'll be good to merge.

Reply:
Okay, I pushed, but I seem to have broken...something. Maybe it's always broken? I'm trying to test to make sure the conflicts work and:

```shell
❯ spack spec -IL mapl@2.42.0 ^openmpi
==> Error: Unknown namespace: CHANGELOG
❯ spack spec -IL mapl@2.42.0 ^mpich@3
==> Error: Unknown namespace: CHANGELOG
❯ spack spec -IL mapl@2.42.0 ^mpich@4
==> Error: Unknown namespace: CHANGELOG
```

I'm really confused. Not sure how I broke this. Am I specifying MPI wrong?